cpp/inc/bond/ext/grpc/client_callback.h

```diff
@@ -0,0 +1,41 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#pragma once
+
+#include <grpc++/impl/codegen/status.h>
+#include <grpc++/client_context.h>
+
+#include <memory>
+
+namespace bond { namespace ext { namespace gRPC {
+
+    template <typename TResponse>
+    struct client_callback_args
+    {
+        bond::bonded<TResponse> response;
+        grpc::Status status;
+        std::shared_ptr<grpc::ClientContext> context;
+
+        explicit client_callback_args(std::shared_ptr<grpc::ClientContext> context)
```

Comment:
When the proxy is creating its `client_unary_call_data` instance, which has a `client_callback_args` inside of it, it only has a ClientContext. Later, when reading the response, the `response` and `status` members will be updated in-place.